### PR TITLE
fix(hardfork): integration fixes

### PR DESCRIPTION
**Pull request by mtsitrin**
# PR Standards

## Opening a pull request should be able to meet the following requirements

--

PR naming convention: https://hackmd.io/@nZpxHZ0CT7O5ngTp0TP9mg/HJP_jrm7A

---

Close #XXX

<-- Briefly describe the content of this pull request -->

For Author:

- [ ]  Targeted PR against correct branch
- [ ]  included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Linked to Github issue with discussion and accepted design
- [ ]  Targets only one github issue
- [ ]  Wrote unit and integration tests
- [ ]  All CI checks have passed
- [ ]  Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code)

---

For Reviewer:

- [ ]  confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Reviewers assigned
- [ ]  confirmed all author checklist items have been addressed

---

After reviewer approval:

- [ ]  In case targets main branch, PR should be squashed and merged.
- [ ]  In case PR targets a release branch, PR should be rebased.

### CODE REVIEW

**Review by jgimeno — APPROVED** [review]

File: block/produce.go [hunk 1]
```diff
@@ -41,7 +41,7 @@ func (m *Manager) ProduceBlockLoop(ctx context.Context, bytesProducedC chan int)
 			return nil
 		case <-ticker.C:
 			// Only produce if I'm the current rollapp proposer.
-			if !m.AmIProposerOnRollapp() {
+			if m.RunMode != RunModeProposer {
```
Comment (srene) [inline block/produce.go:44]: why is this modified? is this runMode not always be the same after start? i mean if the produceblockloop is started it already means is the proposer... and the runmode is not updated after start i think.

Comment (mtsitrin) [reply block/produce.go:44]: so no reason to check against local keys each time again

Comment (omritoptix) [reply block/produce.go:44]: @mtsitrin this is for when you rotate. 

during rotation you start as the proposer, but you shouldn't produce blocks after you produced your last block yet you should still send your last batch. 

the code you introduced will make the proposer continue producing blocks while he's submitting his last batch in parallel to the new proposer.
